src/main/java/com/github/maxopoly/MemeMana/model/MemeManaPouch.java

```diff
@@ -28,9 +37,13 @@ public void cleanupPouch() {
 		}
 	}
 
-	public void addNewUnit(int id, double amount) {
+	public void addNewUnit(MemeManaUnit unit) {
 		// TODO mirror in db
-		units.add(new MemeManaUnit(id, amount));
+		units.add(unit);
+	}
+
+	public List<MemeManaUnit> getUnits() {
```

Comment:
The internal units should not be exposed like that, it heavily violates encapsulation. If you need to do something on them (like adding an item), add a method to this class to sort it in, but dont give out a reference to the list used internally.

Reply:
Both usages of this are fixed by sufficiently powerful SQL queries, so this is moot.